test/common/sink_test.cc

```diff
@@ -0,0 +1,138 @@
+#include <filesystem>
+#include <fstream>
+
+#include "external/envoy/source/common/common/random_generator.h"
+
+#include "common/sink_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace {
+
+// Future sink implementations register here for testing top level generic sink behavior.
+using SinkTypes = ::testing::Types<FileSinkImpl>;
+
+template <typename T> class TypedSinkTest : public ::testing::Test {
+public:
+  void SetUp() override { uuid_ = random_.uuid(); }
+  std::string executionIdForTest() const { return uuid_; }
+
+private:
+  Envoy::Random::RandomGeneratorImpl random_;
+  std::string uuid_;
+};
+
+TYPED_TEST_SUITE(TypedSinkTest, SinkTypes);
+
+TYPED_TEST(TypedSinkTest, BasicSaveAndLoad) {
+  TypeParam sink;
+  nighthawk::client::ExecutionResponse result_to_store;
+  *(result_to_store.mutable_execution_id()) = this->executionIdForTest();
+  absl::Status status = sink.StoreExecutionResultPiece(result_to_store);
+  ASSERT_EQ(status.ok(), true);
```

Comment:
nit: If `ASSERT_TRUE` / `ASSERT_FALSE` are available, it seems more standard (here and below).

Reply:
done: 00ebf87